src/protagonist/API/Features/DeliveryChannels/Requests/DeleteCustomerDefaultDeliveryChannel.cs

```diff
@@ -0,0 +1,53 @@
+﻿using DLCS.Core;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.DeliveryChannels.Requests;
+
+public class DeleteCustomerDefaultDeliveryChannel : IRequest<ResultMessage<DeleteResult>>
+{
+    public DeleteCustomerDefaultDeliveryChannel(int customer, string defaultDeliveryChannelId)
+    {
+        Customer = customer;
+        DefaultDeliveryChannelId = defaultDeliveryChannelId;
+    }
+    
+    public int Customer { get; }
+    
+    public string DefaultDeliveryChannelId { get; }
+}
+
+public class DeleteCustomHeaderHandler : IRequestHandler<DeleteCustomerDefaultDeliveryChannel, ResultMessage<DeleteResult>>
+{
+    private readonly DlcsContext dbContext;
+
+    public DeleteCustomHeaderHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+
+    public async Task<ResultMessage<DeleteResult>> Handle(DeleteCustomerDefaultDeliveryChannel request, CancellationToken cancellationToken)
+    {
+        var isGuid = Guid.TryParse(request.DefaultDeliveryChannelId, out var defaultDeliveryChannelGuid);
```

Comment:
This applied in a couple of other places too but rather than validate this here would it be easier to specify that restriction via a [route constraint](https://learn.microsoft.com/en-us/aspnet/web-api/overview/web-api-routing-and-actions/attribute-routing-in-web-api-2#route-constraints)? A non-guid value would return a 404 which I think is okay.

`[HttpDelete("{defaultDeliveryChannelId:guid}")]`

Alternatively can you pass a `guid` to an action-method, rather than a string?